### PR TITLE
Unicode slugs for Django 1.9

### LINKED ISSUE #1994: Unicode slugs
Since `SlugField` supports unicode since Django 1.9 (https://docs.djangoproject.com/es/1.9/ref/models/fields/#django.db.models.SlugField.allow_unicode), which is already supported by latest Oscar, we should probably think about proper unicode slugs support in the Oscar too.

Firstly, we could make optional conversion from unicode to ASCII in the slugify function - https://github.com/django-oscar/django-oscar/blob/master/src/oscar/core/utils.py#L41.

Also, perhaps there's no need in custom `oscar.models.fields.AutoSlugField`, since original one from `django_extensions` already support custom slugify function - https://github.com/django-extensions/django-extensions/blob/master/django_extensions/db/fields/__init__.py#L109.

Thanks to @pwalsh for the help in discovering this issue.

### PR DESCRIPTION
**Pull request by sasha0**
Fixes #1994 

Initial feature design has trade-offs in favor of backwards compatibility.
Improvement suggestions welcomed.

### CODE REVIEW

**Comment by pwalsh** [comment]
Would be great to get this looked at. Allows us to move forward with Django 1.9 which (finally!) accepts that people on the web actually do use unicode slugs.

File: src/oscar/models/fields/slugfield.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class SlugField(DjangoSlugField):
+    def __init__(self, *args, **kwargs):
+        # not override parameter if it was passed explicitly,
+        # so passed parameters takes precedence over the setting
+        if getattr(settings, 'OSCAR_SLUG_ALLOW_UNICODE', None) and \
+                'allow_unicode' not in kwargs:
+            kwargs['allow_unicode'] = True
+
```
Comment (mvantellingen) [inline src/oscar/models/fields/slugfield.py]: `kwargs.setdefault('allow_unicode', settings.OSCAR_SLUG_ALLOW_UNICODE)`

File: src/oscar/core/utils.py [hunk 2]
```diff
@@ -39,9 +44,11 @@ def slugify(value):
         slugifier = import_string(slugifier)
 
     # Use unidecode to convert non-ASCII strings to ASCII equivalents where
-    # possible.
-    value = slugifier(unidecode(six.text_type(value)))
+    # possible if unicode is not allowed to contain in slug.
+    if not getattr(settings, 'OSCAR_SLUG_ALLOW_UNICODE', None):
```
Comment (mvantellingen) [inline src/oscar/core/utils.py]: The `settings.OSCAR_SLUG_ALLOW_UNICODE` should always be set right? So better to use it directly (also on the other places)

File: src/oscar/core/utils.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 from django.conf import settings
 from django.shortcuts import redirect, resolve_url
 from django.template.defaultfilters import date as date_filter
-from django.template.defaultfilters import slugify as django_slugify
+from django.utils.text import slugify as django_slugify
```
Comment (mvantellingen) [inline src/oscar/core/utils.py]: import order

File: src/oscar/apps/basket/abstract_models.py [hunk 4]
```diff
@@ -10,6 +10,7 @@
 from django.utils.translation import ugettext_lazy as _
 
 from oscar.apps.basket.managers import OpenBasketManager, SavedBasketManager
+from oscar.models.fields.slugfield import SlugField
```
Comment (mvantellingen) [inline src/oscar/apps/basket/abstract_models.py]: import order

**Comment by mvantellingen** [comment]
Looks really nice @sasha0 ! Few minor things and perhaps you can add some small text in the release notes for 1.2?

**Comment by sasha0** [comment]
Sure, will do.

**Comment by pwalsh** [comment]
Thanks for reviewing @mvantellingen

**Comment by sasha0** [comment]
@mvantellingen I've updated PR, please take a look. Thanks.

**Pull request merged**